test/functional/sqlsrv/sqlsrv_connect_driver.phpt

```diff
@@ -75,35 +78,92 @@ function test_valid_values( $msodbcsql_maj ,$server ,$connectionOptions){
     }
     
     $connectionOptions['Driver']=$value;
-    connect_verify_output( $msodbcsql_maj ,$server ,$connectionOptions );
+    connect_verify_output( $server, $connectionOptions );
 }
 
-function test_invalid_values($msodbcsql_maj ,$server ,$connectionOptions){
+function test_invalid_values( $msodbcsql_maj, $server, $connectionOptions )
+{
     // test invalid value
     $value = "{SQL Server Native Client 11.0}";
     $connectionOptions['Driver']=$value;
     $expected = "Invalid value $value was specified for Driver option.";
-    connect_verify_output( $msodbcsql_maj ,$server ,$connectionOptions ,$expected );
+    connect_verify_output( $server, $connectionOptions, $expected );
     
     $value = "SQL Server Native Client 11.0";
     $connectionOptions['Driver']=$value;
     $expected = "Invalid value $value was specified for Driver option.";
-    connect_verify_output( $msodbcsql_maj ,$server ,$connectionOptions ,$expected );
+    connect_verify_output( $server, $connectionOptions, $expected );
     
     $value = "ODBC Driver 00 for SQL Server";
     $connectionOptions['Driver']=$value;
     $expected = "Invalid value $value was specified for Driver option.";
-    connect_verify_output( $msodbcsql_maj ,$server ,$connectionOptions ,$expected );
+    connect_verify_output( $server, $connectionOptions, $expected );
     
     $value = 123;
     $connectionOptions['Driver']=$value;
     $expected = "Invalid value type for option Driver was specified.  String type was expected.";
-    connect_verify_output( $msodbcsql_maj ,$server ,$connectionOptions ,$expected );
+    connect_verify_output( $server, $connectionOptions, $expected );
     
     $value = false;
     $connectionOptions['Driver']=$value;
     $expected = "Invalid value type for option Driver was specified.  String type was expected.";
-    connect_verify_output( $msodbcsql_maj ,$server ,$connectionOptions ,$expected );
+    connect_verify_output( $server, $connectionOptions, $expected );
```

Comment:
same, please use array of values and iterate over it

Reply:
For pdo test we could do that but in sqlsrv the expected errors are not all the same.